File: 프로그래머스/1/133499. 옹알이 （2）/옹알이 （2）.js
function solution(babbling) {
    let answer = 0;    

    for (let i = 0; i < babbling.length; i++) {

        let isCheck = true;
        let babyStrDuplicationCheck = 0;

        for (let j = 0; j < babbling[i].length; j++) {           

            if (babbling[i].substr(j, 3) == "aya" && babyStrDuplicationCheck != 1) {
                babyStrDuplicationCheck = 1;
                j += 2;
            }
            else if (babbling[i].substr(j, 2) == "ye" && babyStrDuplicationCheck != 2) {
                babyStrDuplicationCheck = 2;
                j += 1;
            }
            else if (babbling[i].substr(j, 3) == "woo" && babyStrDuplicationCheck != 3) {
                babyStrDuplicationCheck = 3;
                j += 2;
            }
            else if (babbling[i].substr(j, 2) == "ma" && babyStrDuplicationCheck != 4) {
                babyStrDuplicationCheck = 4;
                j += 1;
            }
            else {
                isCheck = false;
                break;
            }            
        }

        if (isCheck == true) {
            answer++;
        }
    }

    return answer;
}

File: 프로그래머스/1/133499. 옹알이 （2）/옹알이 （2）.cpp
#include <string>
#include <vector>

using namespace std;

int solution(vector<string> babbling) {
    int answer = 0;
    bool isCheck = false;

    for (int i = 0; i < babbling.size(); i++)
    {
        isCheck = true;

        for (int j = 0; j < babbling[i].size(); j++)
        {
            if (babbling[i].substr(j, 3) == "aya")
            {
                j += 2;
            }
            else if (babbling[i].substr(j, 2) == "ye")
            {
                j += 1;
            }
            else if (babbling[i].substr(j, 3) == "woo")
            {
                j += 2;
            }
            else if (babbling[i].substr(j, 2) == "ma")
            {
                j += 1;
            }
            else
            {
                isCheck = false;
                break;
            }
        }
        if (isCheck == true)
        {
            answer++;
        }
    }

    return answer;
}